### ISSUE TITLE
Unify / Deduplicate the GitHub cache

### ISSUE THREAD

**Issue by decentralion** [sourcecred/sourcecred#1119]
Consider the following sequence of commands:

```bash
$ sourcecred load org/big-repo org/small-repo-foo  --output org/big-with-foo
$ sourcecred load org/big-repo org/small-repo-bar  --output org/big-with-bar
```

One might reasonably suppose that since we cache results from GitHub (cf: #622), we will only need to fetch `org/big-repo` once, and share it across both invocations.

However, as implemented, the cache is maintained per output repository, meaning that the big-repo must be fetched twice.

cc/ @wchargin for comment. How difficult is this to fix? Can we add the good-first-issue label?

Note that work here might be superceded if we find a good way to represent projects as first-class citizens, and do away with the (hacky) --output flag (cf: #701)

**Comment by wchargin** [comment]
As you suggest, the problem here is in the core layer, not the GitHub
plugin: the GitHub plugin isn’t afforded the opportunity to share cache
in the way that it needs to.

Perhaps one fix is to just provide plugins with a top-level cache
directory, for which they are expected to manage repositories
themselves? (The GitHub plugin will already do this correctly, fwiw,
because the databases are called `mirror_*.db` where the `*` component
is an encoding of the repo ID.)

**Comment by decentralion** [comment]
Fixed by #1233
